llpc/test/shaderdb/general/PipelineVsFs_TestPrimitiveID_Last.pipe

```diff
@@ -0,0 +1,78 @@
+; Test primitiveID when provoking vertex mode is LAST
+
+; BEGIN_SHADERTEST
+; RUN: amdllpc -spvgen-dir=%spvgendir% -v --gfxip=10.1.0 %s | FileCheck -check-prefix=SHADERTEST %s
+; SHADERTEST-LABEL: {{^// LLPC}} pipeline patching results
+; SHADERTEST: @[[LDS:[^ ]*]] = external addrspace(3) global [{{[0-9]*}} x i32], align 4
+; SHADERTEST: define dllexport amdgpu_gs void @_amdgpu_gs_main(i32
+; SHADERTEST: %{{[0-9]*}} = lshr i32 %{{[^,]*}}, 18
```

Comment:
This still looks odd to me.  I'm not sure why you need to match this instruction.  The input register could be anything, and the result does not have to be used.  It seems to provide nothing to the store being done below.

I'm guessing this result of this instruction is used in the next one.  If so, that should be made explicit.  Also, it might be useful to explain what the 18 and 2044 are suppose to do in a comment:

```
; Extract bits n to m from "meaningful name for input".  These bit encode the <explain what it is> which will be used to 
; calculate the address at which to store the provoking vertex.
; SHADERTEST: [[something:%[0-9]*]] = lshr i32 [[type]] [[meaningful_name_for_input]] , 18
; SHADERTEST: [[offset:%[0-9]*]] = and i32 [[something]], 2044
```

This would help somebody like me understand what is going on, and call out that this 18 is the important difference with the other test.

Reply:
Thanks for your guidance. Learned.